core-bundle/src/Resources/contao/library/Contao/Search.php

```diff
@@ -592,12 +719,24 @@ public static function removeEntry($strUrl)
 
 		while ($objResult->next())
 		{
+			// Decrement document frequency counts
+			$objDatabase
+				->prepare("
+					UPDATE tl_search_term
+					INNER JOIN tl_search_index ON tl_search_term.id = tl_search_index.termId AND tl_search_index.pid = ?
+					SET documentFrequency = GREATEST(0, documentFrequency - 1)
+				")
+				->execute($objResult->id);
+
 			$objDatabase->prepare("DELETE FROM tl_search WHERE id=?")
 						->execute($objResult->id);
 
 			$objDatabase->prepare("DELETE FROM tl_search_index WHERE pid=?")
 						->execute($objResult->id);
 		}
+
+		// Remove obsolete terms
+		$objDatabase->query("DELETE FROM tl_search_term WHERE documentFrequency = 0");
```

Comment:
Should we have an index on `documentFrequency` for this purpose?

Reply:
I don’t think that is necessary, this query is already very fast IMO.

Reply:
Ok :)